### PR TITLE
Added Stream and release input options

### PR DESCRIPTION
**Pull request by phillip-kruger**
As optional parameters in the REST Endpoints

### CODE REVIEW

**Comment by phillip-kruger** [comment]
swagger1
swagger2
swagger3

At the moment, the OpenAPI schema document builds at startup, and when a new release is created, the scheduler will pick it up create the new catalog, but the OpenAPI schema will now update until we restart. I have created https://github.com/smallrye/smallrye-open-api/issues/837 to solve this. This only effect the schema, and a user will still be able to call the rest endpoint with the new release input.

**Comment by gsmet** [comment]
Just a note that if this needs the latest changes made in `main`, it shouldn't be merged yet!

File: src/main/kotlin/io/quarkus/code/rest/CodeQuarkusResource.kt [hunk 1]
```diff
@@ -79,30 +90,126 @@ class CodeQuarkusResource {
...
+        description = "All available platforms",
+        content = [Content(
+            mediaType = APPLICATION_JSON,
+            schema = Schema(implementation = PlatformCatalog::class)
+        )]
+    )
+    fun platforms(): Response {
+        val platformCatalog = platformService.platformCatalog
+        val lastUpdated = platformService.lastUpdated
+        return Response.ok(platformCatalog).header(LAST_MODIFIED_HEADER,lastUpdated.format(formatter) + " GMT").build()
```
[the remark this answers is not in the document]
Comment (phillip-kruger) [reply src/main/kotlin/io/quarkus/code/rest/CodeQuarkusResource.kt]: Done, Thanks :)

**Comment by phillip-kruger** [comment]
As discussed, I removed the "catalogForRelease" Endpoint for now.

**Pull request closed without merging**